File: kotlin/src/main/kotlin/dayOne/1.kt
package dayOne
import java.io.File

class ExpenseReport {
    private val fileName = System.getProperty("user.dir") + "/src/main/kotlin/dayOne/input.txt"
    private val expenseList: List<Int> = File(fileName).readLines().map { Integer.parseInt(it) }


    fun getProduct(): Int? {
        for (multiplicand in expenseList) {
            for (multiplier in expenseList) {
                if (multiplicand + multiplier == 2020) {
                    return multiplicand * multiplier
                }
            }
        }
        return null
    }


    fun getProductPart2(): Int? {
        for (first in expenseList) {
            for (second in expenseList) {
                for (third in expenseList) {
                    if (first + second + third == 2020) {
                        return first * second * third
                    }
                }
            }
        }
        return null
    }

}


fun main() {
    val report = ExpenseReport()
    val answer1 = report.getProduct()
    val answer2 = report.getProductPart2()

    println("Answer 1: $answer1")
    println("Answer 2: $answer2")
}

File: python/1/1.py
from typing import Optional
from common import readList

def get2020Product() -> Optional[int]:
    expense_list = readList('input.txt', int)
    for multiplicand in expense_list:
        for multiplier in expense_list:
            if multiplicand + multiplier == 2020:
                return multiplicand * multiplier
    return None


def get2020ProductPart2() -> Optional[int]:
    expense_list = readList('input.txt', int)
    for first in expense_list:
        for second in expense_list:
            for third in expense_list:
                if first + second + third == 2020:
                    return first * second * third
    return None


def main():
    answer1 = get2020Product()
    answer2 = get2020ProductPart2()

    print(f'Answer 1: {answer1}')
    print(f'Answer 2: {answer2}')


if __name__ == '__main__':
    main()
